frontend/components/app.js

```diff
@@ -1,35 +1,16 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import Notifications from 'react-notification-system-redux';
+import { connect } from 'react-redux';
 
 import alertActions from '../actions/alertActions';
 import LoadingIndicator from './loadingIndicator';
 import Header from './header';
 
-function getUsername(storeState) {
-  const userState = storeState.user;
-  if (!userState.isLoading && userState.data) {
-    return userState.data.username;
-  }
-  return null;
-}
-
-class App extends React.Component {
-  constructor(props, context) {
-    super(props, context);
-    this.state = this.getStateFromStore();
-  }
-
+export class App extends React.Component {
   componentWillReceiveProps(nextProps) {
-    const state = this.getStateFromStore();
```

Comment:
I'll be happy when we can move everything over to our notification lib and all of this code can go away.